src/main/java/uk/gov/companieshouse/controller/AppealController.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.companieshouse.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.headers.Header;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import uk.gov.companieshouse.model.Appeal;
+import uk.gov.companieshouse.service.AppealService;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/companies")
+public class AppealController {
+
+    private static final Logger LOGGER = org.slf4j.LoggerFactory.getLogger(AppealController.class);
+
+    private final AppealService appealService;
+
+    public AppealController(AppealService appealService) {
+        this.appealService = appealService;
+    }
+
+    @Operation(summary = "Create a new appeal", tags = "Appeal")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "201", description = "Appeal resource created", headers = {
+            @Header(name = "location")
+        }),
+        @ApiResponse(responseCode = "400", description = "Bad request"),
+        @ApiResponse(responseCode = "401", description = "Unauthorised request"),
+        @ApiResponse(responseCode = "422", description = "Invalid appeal data"),
+        @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @PostMapping(value = "/{company-id}/appeals", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> submitAppeal(@RequestHeader("ERIC-identity") String userId,
+                                               @PathVariable("company-id") final String companyId,
+                                               @Valid @RequestBody final Appeal appeal) {
+
+        final String penaltyReference = appeal.getPenaltyIdentifier().getPenaltyReference();
```

Comment:
It is bit inconsistent. Some variables like here have final modifier others do not.

Reply:
Change if you can, leave if too much work. Not super important at the moment.

Reply:
they all are `final` where they can be